src/mlpack/methods/ann/regularizer/no_regularizer.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * @file no_regularizer.hpp
+ * @author Saksham Bansal
+ *
+ * Definition of the NoRegularizer class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_NOREGULARIZER_HPP
+#define MLPACK_METHODS_ANN_NOREGULARIZER_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann {
+
+/**
+ * Implementation of the NoRegularizer. This does not add any
+ * regularization to the weights.
+ */
+class NoRegularizer
+{
+ public:
+  /**
+   * Create the regularizer object.
+   *
+   * @param factor The factor for regularization.
+   */
+  NoRegularizer()
+  {
+    // Nothing to do here.
+  };
+
+  /**
+   * Calculate the gradient for regularization.
+   *
+   * @tparam MatType Type of weight matrix.
+   * @param weight The weight matrix to be regularized.
+   * @param gradient The calculated gradient.
+   */
+  template<typename MatType>
+  void Evaluate(const MatType& weight, MatType& gradient)
```

Comment:
To avoid unused parameter warning, let's comment `´weight, gradient``:

```c++
void Evaluate(const MatType& /*weight*/, MatType& /*gradient*/)
```